lib/Conversion/ATenToXTenPass.cpp

```diff
@@ -47,33 +47,67 @@
 #define DEBUG_TYPE "aten-to-xten-pass"
 
 using namespace mlir;
-using namespace xilinx::xten;
+// using namespace xilinx::xten;
+using namespace xilinx;
+using namespace mlir::torch;
 
 namespace {
 
+long getInputSize(xten::Conv2dOp &c2d) {
+  TensorType inputType = c2d.input().getType().dyn_cast<Torch::ValueTensorType>().toBuiltinTensor();
+  if (inputType == nullptr || !inputType.hasRank())
+    return 0;
+  auto shape = inputType.getShape();
+  // multiply all dimensions together
+  long result = 1;
+  for (auto it = shape.begin(); it != shape.end(); it++)
+    result *= *it;
```

Comment:
NIT: Range-based for loop is easier to read
```suggestion
  for (auto i: shape)
    result *= i;
```

Though you could also use `std::reduce(shape.begin(), shape.end(), 1, std::multiplies<long>())`

Reply:
Your loop version is nicer than the reduce I think. Thanks for spotting this